src/utils/helper.ts

```diff
@@ -1,11 +1,12 @@
+import { version } from 'react'
 export const noop = () => {}
 
 // Using noop() as the undefined value as undefined can possibly be replaced
 // by something else.  Prettier ignore and extra parentheses are necessary here
 // to ensure that tsc doesn't remove the __NOINLINE__ comment.
 // prettier-ignore
 export const UNDEFINED = (/*#__NOINLINE__*/ noop()) as undefined
-
+export const IS_REACT_LEGACY = !version.startsWith('18')
```

Comment:
experimental versions starts with `0.0.0`, we could check if it has react 18 api
```suggestion
export const IS_REACT_LEGACY = !React.useTransition
```

Reply:
`useTransition` is too long, we can use `useId` for this case :)

Reply:
😂 save byte by byte

Reply:
Interestingly someone just mentioned about Preact compatibility: https://github.com/vercel/swr/discussions/1645#discussioncomment-2578618